### PR TITLE
Migrate to bzlmod

### LINKED ISSUE #5432: Migrate to Bzlmod for managing external dependencies
With future Bazel LTS versions, Bzlmod will become the default system for external dependencies management.

Please consider migrating all your external dependencies in WORKSPACE to MODULE.bazel

For more information, please check https://github.com/bazelbuild/bazel/issues/18958

### PR DESCRIPTION
**Pull request by agluszak**
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [ ] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: `<please reference the issue number or url here>`

# Description of this change
Closes #5432.
Migration to bzlmod will be required at some point. I replaced all `jvm_maven_import_external` with`rules_jvm_external`'s mechanism so that we have a single point for downloading jar dependencies. I bumped versions of some rules - these which had no support for bzlmod.

### CODE REVIEW

File: MODULE.bazel
Comment (agluszak) [reply MODULE.bazel:3]: yup, I think there were some issues without it, but I can try removing it

File: aspect/testing/tests/src/com/google/idea/blaze/aspect/general/noide/NoIdeTest.java
Comment (agluszak) [reply aspect/testing/tests/src/com/google/idea/blaze/aspect/general/noide/NoIdeTest.java:37]: it's the same, containsAllOf was renamed in the new version (which we use now)

File: querysync/javatests/com/google/idea/blaze/qsync/QuerySyncTestUtils.java
Comment (agluszak) [reply querysync/javatests/com/google/idea/blaze/qsync/QuerySyncTestUtils.java:142]: if you prefer it I can turn it into a regex

File: MODULE.bazel
Comment (agluszak) [reply MODULE.bazel:141]: but they were imported with `jvm_maven_import_external`

File: MODULE.bazel
Comment (agluszak) [reply MODULE.bazel:110]: I'm afraid you cannot reference variables in MODULE.bazel without using some workarounds

Comment (agluszak) [reply MODULE.bazel:110]: this should go away once grpc migrates to bzlmod https://github.com/grpc/grpc-java/issues/9559

File: MODULE.bazel
Comment (agluszak) [reply MODULE.bazel:25]: I wanted to update as little things as possible, let's postpone bumping python's version to another PR

File: MODULE.bazel
Comment (agluszak) [reply MODULE.bazel:3]: nope, it has to be there

File: querysync/javatests/com/google/idea/blaze/qsync/testdata/kotlinlib/BUILD
Comment (blorente) [inline querysync/javatests/com/google/idea/blaze/qsync/testdata/kotlinlib/BUILD:1]: Seems like `tools/build_defs/kotlin:rules.bzl` is no longer used. Could we delete it?

File: WORKSPACE.bzlmod
Comment (agluszak) [reply WORKSPACE.bzlmod:1]: > Why not keep both WORKSPACE files, at least for one release? That way we have a fallback in case something goes wrong.
Because then we will have to keep them in sync. If something goes wrong, we can just revert this PR, can't we?

File: querysync/javatests/com/google/idea/blaze/qsync/testdata/kotlinlib/BUILD
Comment (agluszak) [reply querysync/javatests/com/google/idea/blaze/qsync/testdata/kotlinlib/BUILD:1]: it's already gone

File: base/tests/unittests/com/google/idea/blaze/base/qsync/DependencyTrackerImplTest.java
Comment (blorente) [reply base/tests/unittests/com/google/idea/blaze/base/qsync/DependencyTrackerImplTest.java:194]: @mai93 , do you know what this test case was supposed to check?

File: WORKSPACE.bzlmod
Comment (blorente) [reply WORKSPACE.bzlmod:1]: > If something goes wrong, we can just revert this PR, can't we?

Given the number of files this PR touches, it's likely that someone will merge a change that makes the revert difficult at some point. But, you've already done most of the work for this and it's likely only going to impact us working on the plugin, so I'm not going to block the PR.

**Comment by blorente** [comment]
I just tried to import this on my machine, and  found a couple of issues:

- The plugin refuses to import the plugin:
Screenshot 2023-12-19 at 15 01 20
- When I run `bazel test ... --define=ij_product=intellij-ue-under-dev`, I get:
```shell
❯ bazel test ... --define=ij_product=intellij-ue-under-dev
INFO: Invocation ID: 10dc268b-91d6-4f50-b4dd-4e2ea010edd0
WARNING: Build options --define and --java_runtime_version have changed, discarding analysis cache (this can be expensive, see https://bazel.build/advanced/performance/iteration-speed).
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
[1 / 1] checking cached actions
INFO: Repository rules_bazel_integration_test~0.20.0~bazel_binaries~bazel_binaries_bazelisk instantiated at:
  <builtin>: in <toplevel>
Repository rule bazelisk_binary defined at:
  /private/var/tmp/_bazel_blorente/99101ccc32544f31e060bfc1a1f5d4f8/external/rules_bazel_integration_test~0.20.0/bazel_integration_test/private/bazel_binaries.bzl:95:34: in <toplevel>
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
[1 / 1] checking cached actions
ERROR: no such package '@@[unknown repo 'remoteapis' requested from @@]//': The repository '@@[unknown repo 'remoteapis' requested from @@]' could not be resolved: No repository visible as '@remoteapis' from main repository
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
ERROR: /Volumes/code/github.com/bazelbuild/intellij/third_party/bazel/src/main/protobuf/BUILD:147:14: no such package '@@[unknown repo 'remoteapis' requested from @@]//': The repository '@@[unknown repo 'remoteapis' requested from @@]' could not be resolved: No repository visible as '@remoteapis' from main repository and referenced by '//third_party/bazel/src/main/protobuf:remote_execution_log_proto'
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
Use --verbose_failures to see the command lines of failed build steps.
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
ERROR: Analysis of target '//third_party/bazel/src/main/protobuf:remote_execution_log_proto' failed; build aborted: Analysis failed
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
INFO: Elapsed time: 1.107s, Critical Path: 0.00s
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
INFO: 1 process: 1 internal.
Analyzing: 791 targets (163 packages loaded, 607 targets configured)
ERROR: Build did NOT complete successfully
FAILED:
FAILED:
ERROR: No test targets were found, yet testing was requested
FAILED:
    Fetching repository @@rules_java~7.1.0~toolchains~local_jdk; starting
    Fetching https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-darwin-arm64
```

**Comment by agluszak** [comment]
> I just tried to import this on my machine (commit `1931c3e899446c3bbc18`), and found a couple of issues:
> 
>     * The plugin refuses to import the plugin:
> 
> 
> Screenshot 2023-12-19 at 15 01 20
> 
>     * When I run `bazel clean --expunge && bazel test ... --define=ij_product=intellij-ue-under-dev`, I get:
> 
> 
> ```shell
> ❯ bazel test ... --define=ij_product=intellij-ue-under-dev
> INFO: Invocation ID: 10dc268b-91d6-4f50-b4dd-4e2ea010edd0
> WARNING: Build options --define and --java_runtime_version have changed, discarding analysis cache (this can be expensive, see https://bazel.build/advanced/performance/iteration-speed).
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> [1 / 1] checking cached actions
> INFO: Repository rules_bazel_integration_test~0.20.0~bazel_binaries~bazel_binaries_bazelisk instantiated at:
>   <builtin>: in <toplevel>
> Repository rule bazelisk_binary defined at:
>   /private/var/tmp/_bazel_blorente/99101ccc32544f31e060bfc1a1f5d4f8/external/rules_bazel_integration_test~0.20.0/bazel_integration_test/private/bazel_binaries.bzl:95:34: in <toplevel>
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> [1 / 1] checking cached actions
> ERROR: no such package '@@[unknown repo 'remoteapis' requested from @@]//': The repository '@@[unknown repo 'remoteapis' requested from @@]' could not be resolved: No repository visible as '@remoteapis' from main repository
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> ERROR: /Volumes/code/github.com/bazelbuild/intellij/third_party/bazel/src/main/protobuf/BUILD:147:14: no such package '@@[unknown repo 'remoteapis' requested from @@]//': The repository '@@[unknown repo 'remoteapis' requested from @@]' could not be resolved: No repository visible as '@remoteapis' from main repository and referenced by '//third_party/bazel/src/main/protobuf:remote_execution_log_proto'
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> Use --verbose_failures to see the command lines of failed build steps.
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> ERROR: Analysis of target '//third_party/bazel/src/main/protobuf:remote_execution_log_proto' failed; build aborted: Analysis failed
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> INFO: Elapsed time: 1.107s, Critical Path: 0.00s
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> INFO: 1 process: 1 internal.
> Analyzing: 791 targets (163 packages loaded, 607 targets configured)
> ERROR: Build did NOT complete successfully
> FAILED:
> FAILED:
> ERROR: No test targets were found, yet testing was requested
> FAILED:
>     Fetching repository @@rules_java~7.1.0~toolchains~local_jdk; starting
>     Fetching https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-darwin-arm64
> ```

The first one is trivial, I forgot to leave an empty WORKSPACE file. I'm investigating the second one now

**Comment by agluszak** [comment]
@blorente has `bazel build ...` ever worked in this repo? When I try running it on master I get this:

```
➜  old-bazel-plugin git:(master) bazel build ... --define=ij_product=intellij-ue-under-dev 
WARNING: Target pattern parsing failed.
ERROR: Skipping '../..': invalid target name '../..': target names may not contain up-level references '..'
ERROR: invalid target name '../..': target names may not contain up-level references '..'
INFO: Elapsed time: 0.054s
INFO: 0 processes.
ERROR: Build did NOT complete successfully
➜  old-bazel-plugin git:(master) bazel build //third_party/bazel/src/main/protobuf/... --define=ij_product=intellij-ue-under-dev
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:action_cache_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:android_deploy_info_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:bazel_flags_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:build_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:builtin_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:crosstool_config_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:deps_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:desugar_deps_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:execution_statistics_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:extra_actions_base_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:invocation_policy_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:java_compilation_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:memory_pressure_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:test_status_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:worker_protocol_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:execution_graph_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:analysis_v2_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:cache_salt_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:command_line_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:command_server_java_grpc_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:command_server_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:failure_details_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:option_filters_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:profile_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:remote_execution_log_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:remote_scrubbing_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:spawn_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: //third_party/bazel/src/main/protobuf:dist_jars: missing input file '//third_party/bazel/src/main/protobuf:xcode_java_proto_srcs'
ERROR: /home/andrzej.gluszak/code/jetbrains/old-bazel-plugin/third_party/bazel/src/main/protobuf/BUILD:218:10: 28 input file(s) do not exist
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.072s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
ERROR: Build did NOT complete successfully
➜  old-bazel-plugin git:(master) bazel build //... --define=ij_product=intellij-ue-under-dev
WARNING: Target pattern parsing failed.
ERROR: Skipping '//...': error loading package under directory '': error loading package 'querysync/javatests/com/google/idea/blaze/qsync/testdata/kotlinlib': Label '//tools/build_defs/kotlin:rules.bzl' is invalid because 'tools/build_defs/kotlin' is not a package; perhaps you meant to put the colon here: '//:tools/build_defs/kotlin/rules.bzl'?
ERROR: error loading package under directory '': error loading package 'querysync/javatests/com/google/idea/blaze/qsync/testdata/kotlinlib': Label '//tools/build_defs/kotlin:rules.bzl' is invalid because 'tools/build_defs/kotlin' is not a package; perhaps you meant to put the colon here: '//:tools/build_defs/kotlin/rules.bzl'?
INFO: Elapsed time: 0.060s
INFO: 0 processes.
ERROR: Build did NOT complete successfully

```

**Comment by blorente** [comment]
@agluszak Fair enough. This is a command line we run in our pre-submit CI, so I'd expect it to work:
```shell
$  bazel test --define=ij_product=clion-latest //:clwb_tests

INFO: Repository rules_java~7.1.0~toolchains~remote_java_tools instantiated at:
  <builtin>: in <toplevel>
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_blorente/99101ccc32544f31e060bfc1a1f5d4f8/external/bazel_tools/tools/build_defs/repo/http.bzl:384:31: in <toplevel>
ERROR: no such package '@@[unknown repo 'io_netty_netty_transport_classes_kqueue' requested from @@]//jar': The repository '@@[unknown repo 'io_netty_netty_transport_classes_kqueue' requested from @@]' could not be resolved: No repository visible as '@io_netty_netty_transport_classes_kqueue' from main repository
ERROR: /Volumes/code/github.com/bazelbuild/intellij/ext/BUILD:9:13: no such package '@@[unknown repo 'io_netty_netty_transport_classes_kqueue' requested from @@]//jar': The repository '@@[unknown repo 'io_netty_netty_transport_classes_kqueue' requested from @@]' could not be resolved: No repository visible as '@io_netty_netty_transport_classes_kqueue' from main repository and referenced by '//ext:intellijext'
ERROR: Analysis of target '//skylark:unit_tests' failed; build aborted: Analysis failed
INFO: Elapsed time: 5.183s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
ERROR: Build did NOT complete successfully
ERROR: No test targets were found, yet testing was requested
FAILED:
    Fetching repository @@bazel_tools~cc_configure_extension~local_config_cc; starting
    Fetching repository @@rules_cc; starting
    Fetching repository @@com_google_absl; starting

```

**Comment by agluszak** [comment]
it's a macos vs linux thingy, because on my (linux) machine it works... let me know if the latest commit fixes it for you

**Comment by blorente** [comment]
@agluszak Yep, commit `799e06f8e1` seemed to do the trick! Thanks for looking into it :D 

I think this surfaces how nice it would be to have macOS CI, but that's for another day. Thanks for putting in this much effort.

**Review by blorente — APPROVED** [review]

File: WORKSPACE.bzlmod
Comment (agluszak) [reply WORKSPACE.bzlmod:618]: so previously something (I don't remember/didn't check) loaded `com_google_protobuf` as a transitive dependency and after I switched that something to bzlmod it became inaccessible

File: querysync/javatests/com/google/idea/blaze/qsync/testdata/TestData.java
Comment (agluszak) [reply querysync/javatests/com/google/idea/blaze/qsync/testdata/TestData.java:57]: hmm, adding 
```
module(
    name = "intellij_with_bazel",
    repo_name = "intellij_with_bazel",
)
```
to MODULE.bazel doesn't seem to affect it

File: aspect/testing/tests/src/com/google/idea/blaze/aspect/java/pluginprocessorjars/BazelPluginProcessorJarTest.java
Comment (mai93) [inline aspect/testing/tests/src/com/google/idea/blaze/aspect/java/pluginprocessorjars/BazelPluginProcessorJarTest.java:37]: I'm not sure if this is the right approach or we should update the aspect to correctly strip the new external dependencies path prefix. we need to look more into why it is needed.

**Review by mai93 — COMMENTED** [review]
I also tried using the plugin built with this change to import the simple java project defined in https://github.com/bazelbuild/intellij/blob/master/examples/java/greetings_project/hello/hello.bazelproject and the initial build failed with:

```
Error: Unable to initialize main class com.google.idea.blaze.aspect.PackageParser
Caused by: java.lang.UnsupportedClassVersionError: com/google/protobuf/MessageLite has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```
It is working fine on the commit right before this.

**Comment by mai93** [comment]
> Oh no, I just noticed windows build doesn't work, but maybe we could address it later. It looks like I hit this problem [protocolbuffers/protobuf#12947](https://github.com/protocolbuffers/protobuf/issues/12947)
> 
> EDIT:
> 
> I was able to workaround it by just shortening the output_base path so NVM

is this considered a regression for using the plugin on windows or do you mean developing the plugin on windows?

File: querysync/javatests/com/google/idea/blaze/qsync/testdata/TestData.java
Comment (mai93) [reply querysync/javatests/com/google/idea/blaze/qsync/testdata/TestData.java:57]: I think this is related to https://github.com/bazelbuild/bazel/issues/18128

**Pull request merged**
